lib/mutant/ast/regexp.rb

```diff
@@ -32,6 +32,23 @@ def self.to_ast(expression)
       def self.to_expression(node)
         Transformer.lookup(node.type).to_expression(node)
       end
+
+      # Convert's a `parser` `regexp` node into more fine-grained AST nodes.
+      #
+      # @param node [Parser::AST::Node]
+      #
+      # @return [Parser::AST::Node]
+      def self.expand_regexp_ast(node)
+        *body, _opts = node.children
```

Comment:
Note that destructuring instead of using `#slice` avoids the mutation [we chose to ignore before](https://github.com/mbj/mutant/pull/1201/files#diff-18b52ac8d1d29dd0bd12125e2a4e496ad0743ab551377dbf443ec10d1f37c991L15).

Reply:
As a followup we could also promote interpolations.

```
a.match?(/\Afoo#{dynamic}/)
```

to:

```
a.end_with?(dynamic)
```

Reply:
Yeah, something like that might be interesting but I am not sure I will go that deep right away. I have a couple of other small improvements for similar cases I am thinking of as well that are pending.